src/hardware/mame/saa1099.cpp

```diff
@@ -256,16 +262,20 @@ void saa1099_device::sound_stream_update(MAYBE_UNUSED sound_stream &stream,
 		for (ch = 0; ch < 6; ch++)
 		{
 			if (m_channels[ch].freq == 0.0)
-				m_channels[ch].freq = (double)((2 * m_master_clock / 512) << m_channels[ch].octave) /
-					(511.0 - (double)m_channels[ch].frequency);
+				m_channels[ch].freq =
+				        (double)((2 * m_baseline_clock / 512)
+				                 << m_channels[ch].octave) /
+				        (511.0 - (double)m_channels[ch].frequency);
```

Comment:
It's the same calculation in here and in line 275 - maybe create a lambda function to avoid code repetition?